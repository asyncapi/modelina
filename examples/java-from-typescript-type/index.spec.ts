const spy = jest.spyOn(global.console, 'log').mockImplementation(() => { return; });
import {generate, generateWithOptions} from './index';

describe('Should be able to generate Java model from a TypeScript type file', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('and should log expected output to console', async () => {
    await generate();
    //Generate is called 6x, so even though we expect 1 model, we double it
    expect(spy.mock.calls.length).toEqual(6);
    expect(spy.mock.calls[1]).toMatchSnapshot();
  });
});

describe(`Should be able to generate Java models from a TypeScript type file 
along with specific options`, () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('and should log expected output to console', async () => {
    await generateWithOptions();
    //GenerateWithOptions is called 6x, so even though we expect 1 model, we double it
    expect(spy.mock.calls.length).toEqual(6);
    expect(spy.mock.calls[1]).toMatchSnapshot();
  });
});
