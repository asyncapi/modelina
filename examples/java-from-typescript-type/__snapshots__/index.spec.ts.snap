// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to generate Java model from a TypeScript type file and should log expected output to console 1`] = `
Array [
  "public class InnerData {
  private Double age;
  private String name;
  private Boolean free;
  private Map<String, Object> additionalProperties;

  public Double getAge() { return this.age; }
  public void setAge(Double age) { this.age = age; }

  public String getName() { return this.name; }
  public void setName(String name) { this.name = name; }

  public Boolean getFree() { return this.free; }
  public void setFree(Boolean free) { this.free = free; }

  public Map<String, Object> getAdditionalProperties() { return this.additionalProperties; }
  public void setAdditionalProperties(Map<String, Object> additionalProperties) { this.additionalProperties = additionalProperties; }
}",
]
`;
